### PR TITLE
test: 35% coverage of view-team-advance-request page

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 650f58c</samp>

This pull request adds a feature to view the receipts of a team advance request, fixes a navigation bug on the view team advance request page, and adds unit tests for the `ViewTeamAdvanceRequestPage` component. The changes affect the files `src/app/fyle/view-team-advance-request/view-team-advance-request.page.ts`, `src/app/fyle/view-team-advance-request/view-team-advance-request.page.html`, and `src/app/fyle/view-team-advance-request/view-team-advance-request.page.spec.ts`.

### <samp>🤖 Generated by Copilot at 650f58c</samp>

> _To fix a bug with the back button_
> _We changed the `routerLink` function_
> _We added a `(click)`_
> _To call `goToTeamAdvances` quick_
> _And used the `router` for navigation_

### <samp>🤖 Generated by Copilot at 650f58c</samp>

*  Fix back button navigation on view team advance request page by replacing `routerLink` with `(click)` event handler that calls `goToTeamAdvances` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-595c90198cc6de49bbc73c3b669f394cdb2504c2494de3bd8c6203db35e37497L8-R8), [link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163R430-R433))
* Add unit tests for `ViewTeamAdvanceRequestPage` component in `src/app/fyle/view-team-advance-request/view-team-advance-request.page.spec.ts` file ([link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-207a312b9a332e35779d4ad67cb5c45fa712eac9aef5b6537d366d5c27189780L1-R228))
* Add `getAttachedReceipts` method to `ViewTeamAdvanceRequestPage` component to fetch and process file objects attached to advance request ([link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163R97-R114))
* Refactor `attachedFiles$` property to use `getAttachedReceipts` method instead of repeating the same logic ([link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L102-R123))
* Move `attachedFiles$` property after `activeApprovals$` property to group properties that depend on `advanceRequest$` property together ([link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L119-R140))
* Remove trailing commas from function or method declarations and calls to follow code style and formatting guidelines ([link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L85-R85), [link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L111-R129), [link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L153-R157), [link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L167-R174), [link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L334-R338), [link](https://github.com/fylein/fyle-mobile-app/pull/2492/files?diff=unified&w=0#diff-8d004d86bbec40f27d3b43a7cfd65c85f287e94158268e2535968aa751638163L376-R380))

## Clickup
https://app.clickup.com/t/85zu314xk

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/view-team-advance-request/view-team-advance-request.page.spec.ts [hunk 1]
```diff
@@ -1,26 +1,229 @@
...
+      'delete',
+      'approve',
+      'sendBack',
+      'reject',
+    ]);
+    const fileServiceSpy = jasmine.createSpyObj('FileService', [
+      'findByAdvanceRequestId',
+      'downloadUrl',
+      'getReceiptsDetails',
+      '',
```
Comment (OmkarJ13) [inline src/app/fyle/view-team-advance-request/view-team-advance-request.page.spec.ts:63]: What is this 😂  ''

File: src/app/fyle/view-team-advance-request/view-team-advance-request.page.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ export class ViewTeamAdvanceRequestPage implements OnInit {
 
   attachedFiles$: Observable<File[] | FileObject[]>;
 
-  advanceRequestCustomFields$: Observable<CustomField[]>;
+  advanceRequestCustomFields$: Observable<CustomField[] | AdvanceRequestsCustomFields[]>;
```
Comment (OmkarJ13) [inline src/app/fyle/view-team-advance-request/view-team-advance-request.page.ts:49]: Why union type here? Can we just use `AdvanceRequestsCustomFields[]?`

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/view-team-advance-request/view-team-advance-request.page.spec.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/fyle/view-team-advance-request/view-team-advance-request.page.spec.ts:63]: 🤣 removing this!

File: src/app/fyle/view-team-advance-request/view-team-advance-request.page.ts (hunk 2 above)
Comment (suyashpatil78) [reply src/app/fyle/view-team-advance-request/view-team-advance-request.page.ts:49]:
```
    const customFields$ = forkJoin({
      advanceRequest: this.advanceRequest$.pipe(take(1)),
      customFields: this.customFields$,
      eou: from(this.authService.getEou()),
    }).pipe(
      map((res: { advanceRequest: ExtendedAdvanceRequest; customFields: CustomField[]; eou: ExtendedOrgUser }) => {
        if (res.eou.ou.org_id === res.advanceRequest.ou_org_id) {
          let customFieldValues = [];
          if (
            res.advanceRequest.areq_custom_field_values !== null &&
            res.advanceRequest.areq_custom_field_values.length > 0
          ) {
            customFieldValues = this.advanceRequestService.modifyAdvanceRequestCustomFields(
              JSON.parse(res.advanceRequest.areq_custom_field_values) as CustomField[]
            );
          }

          res.customFields.map((customField) => {
            customFieldValues.filter((customFieldValue: CustomField) => {
              if (customField.id === customFieldValue.id) {
                customField.value = customFieldValue.value;
              }
            });
          });
          return res.customFields;
        } else {
          return this.advanceRequestService.modifyAdvanceRequestCustomFields(
            JSON.parse(res.advanceRequest.areq_custom_field_values) as CustomField[]
          );
        }
      })
    );
    this.advanceRequestCustomFields$ = customFields$;
```
The customFields$ property is having a mixed type `customFields$: Observable<CustomField[] | AdvanceRequestsCustomFields[]>;` so therefore had to use mixed type here as well.

**Review by shishirams222 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**